api/lib/opentelemetry/distributed_context/propagation/trace_parent.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+class OpenTelemetry::DistributedContext::Propagation::TraceParent
+  InvalidFormatError = Class.new(OpenTelemetry::Error)
+  InvalidVersionError = Class.new(OpenTelemetry::Error)
+  InvalidTraceIDError = Class.new(OpenTelemetry::Error)
+  InvalidSpanIDError = Class.new(OpenTelemetry::Error)
+
+  TRACE_PARENT_HEADER = "traceparent"
+  SUPPORTED_VERSION = 0
+  MAX_VERSION = 254
+
+  REGEXP = /^([A-Fa-f0-9]{2})-([A-Fa-f0-9]{32})-([A-Fa-f0-9]{16})-([A-Fa-f0-9]{2})(-.*)?$/.freeze
+  private_constant :REGEXP
+
+  attr_reader :version, :trace_id, :span_id, :flags
+
+  class Flags
+    def initialize(sampled: false)
+      @sampled = sampled
+    end
+
+    def sampled?
+      @sampled
+    end
+
+    def to_s
+      @sampled ? "01" : "00"
+    end
+  end
+
+  def sampled?
+    flags.sampled?
+  end
+
+  def to_s
+    sprintf("00-%032x-%016x-%s", trace_id, span_id, flags.to_s)
```

Comment:
the text seems to indicate you would parse the version, but then output as the IMPLEMENTATION version